### PR TITLE
[presign] Post presign order

### PR DESCRIPTION
**Pull request by anxolin**
# Summary
The goal of this PR is to differentiate between orders that needs to pre-sign and not, in order to call the API.

* These ones that don't support offchain signing, will call the API as we used to: Adds the signature, and the relevant signature scheme
* The ones that don't will use presign. These order will:
   * call the API first, get the order id
   * Then, it will submit a transaction to presign the order (not included in this PR, it will be the next)

Additionally this PR does two things:
* Remove some legacy code for the signing schemes. Now, instead of redefining it, we can just take the enum from the NPM package
* For making it easy to test, this PR sets to false `allowsOffchainSigning` so EOAs will use presign. We can keep it like this until before we want to consolidate with develop.

# To Test

1. Connect EOA
2. Try to swap
3. The order will be sent and should remain pending ---> FOREVER (for now, no match will occur)

### CODE REVIEW

**Review by alfetopito — APPROVED** [review]
Minor comments.

Regarding the flow, at first I thought the order should fail, but it went straight through https://protocol-explorer.dev.gnosisdev.com/rinkeby/orders/0x1f71f94abf52bbd3b838583994ea503ea7906f598f2cb94412249e808307ee075b0abe214ab7875562adee331deff0fe1912fe42612e8d7e

Ok, that's expected I guess

File: src/custom/pages/Swap/SwapMod.tsx [hunk 1]
```diff
@@ -432,12 +432,13 @@ export default function Swap({
...
-      trade?.tradeType === TradeType.EXACT_INPUT ? <Trans>Receive (incl. fee)</Trans> : null,
-    ]
-  }, [trade])
+  const [exactInLabel, exactOutLabel] = useMemo(
+    () => [
+      independentField === Field.OUTPUT && !showWrap && trade ? <Trans>From (incl. fee)</Trans> : null,
+      independentField === Field.INPUT && !showWrap && trade ? <Trans>Receive (incl. fee)</Trans> : null,
+    ],
+    [independentField, showWrap, trade]
+  )
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/custom/pages/Swap/SwapMod.tsx:441]: yep, I addressed it in another PR so i use the original code

File: src/custom/utils/signatures.ts [hunk 2]
```diff
@@ -30,57 +32,30 @@ export interface SignOrderParams {
...
 // serializable, so no BigNumbers
 //  See https://protocol-rinkeby.dev.gnosisdev.com/api/
 export interface OrderCreation extends UnsignedOrder {
-  // TODO: I commented this because I expect the API and contract to follow the same structure for the order data. confirm and delete this comment
-  signature: string // 65 bytes encoded as hex without `0x` prefix. v + r + s from the spec
-  signingScheme: EcdsaSigningScheme // value of
+  signingScheme: SigningScheme // signed method
+
+  // Signature is only used for offchain signed transactions. It can be omitted for pre-signed ones
+  signature?: string // 65 bytes encoded as hex without `0x` prefix. r + s + v from the spec
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/custom/utils/signatures.ts:45]: yep! Will address this. I forgot to remove the optionality! good eye

[the remark this answers is not in the document]
Comment (anxolin) [reply src/custom/utils/signatures.ts:45]: I will address in the tip of the waterfall

[the remark this answers is not in the document]
Comment (anxolin) [reply src/custom/utils/signatures.ts:45]: https://github.com/gnosis/cowswap/pull/1370/commits/d859563d820cac7f0c5a13c68850e53b99092ffc

**Comment by anxolin** [comment]
Merging, to consolidate the waterfall. I still address issue on the top :)

**Pull request merged**